runtime/drivers/azure/azure.go

```diff
@@ -0,0 +1,332 @@
+	client, err := c.getClient(ctx, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a *blob.Bucket.
+	bucketObj, err := azureblob.OpenBucket(ctx, client, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer bucketObj.Close()
+
+	// prepare fetch configs
+	opts := rillblob.Options{
+		GlobMaxTotalSize:      conf.GlobMaxTotalSize,
+		GlobMaxObjectsMatched: conf.GlobMaxObjectsMatched,
+		GlobMaxObjectsListed:  conf.GlobMaxObjectsListed,
+		GlobPageSize:          conf.GlobPageSize,
+		GlobPattern:           conf.url.Path,
+		ExtractPolicy:         conf.extractPolicy,
+	}
+
+	// Permission error will be thrown while iterating the bucket object, default error will be for az login, considering that as primary auth
+	iter, err := rillblob.NewIterator(ctx, bucketObj, opts, c.logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return iter, nil
+}
+
+type sourceProperties struct {
+	Path                  string         `mapstructure:"path"`
+	URI                   string         `mapstructure:"uri"`
+	Extract               map[string]any `mapstructure:"extract"`
+	GlobMaxTotalSize      int64          `mapstructure:"glob.max_total_size"`
+	GlobMaxObjectsMatched int            `mapstructure:"glob.max_objects_matched"`
+	GlobMaxObjectsListed  int64          `mapstructure:"glob.max_objects_listed"`
+	GlobPageSize          int            `mapstructure:"glob.page_size"`
+	url                   *globutil.URL
+	extractPolicy         *rillblob.ExtractPolicy
+}
+
+func parseSourceProperties(props map[string]any) (*sourceProperties, error) {
+	conf := &sourceProperties{}
+	err := mapstructure.Decode(props, conf)
+	if err != nil {
+		return nil, err
+	}
+	if !doublestar.ValidatePattern(conf.Path) {
+		return nil, fmt.Errorf("glob pattern %s is invalid", conf.Path)
+	}
+	url, err := globutil.ParseBucketURL(conf.Path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse path %q, %w", conf.Path, err)
+	}
+	if url.Scheme != "azblob" {
+		return nil, fmt.Errorf("invalid scheme %q in path %q", url.Scheme, conf.Path)
+	}
+
+	conf.url = url
+	return conf, nil
+}
+
+// getClient returns a new azure blob client.
+func (c *Connection) getClient(ctx context.Context, conf *sourceProperties) (*container.Client, error) {
+	var opts *azureblob.ServiceURLOptions
+	if c.config.AllowHostAccess {
+		opts = azureblob.NewDefaultServiceURLOptions()
+	} else {
+		opts = &azureblob.ServiceURLOptions{
+			AccountName:     c.config.Account,
+			StorageDomain:   c.config.StorageDomain,
+			IsCDN:           c.config.IsCDN,
+			IsLocalEmulator: c.config.IsLocalEmulator,
+			SASToken:        c.config.Key,
+		}
+
+		if c.config.Key != "" {
+			opts.SASToken = c.config.Key
+		} else {
+			opts.SASToken = c.config.SASToken
+		}
+	}
+
+	var client *container.Client
+	if opts.SASToken != "" {
+		serviceURL, err := azureblob.NewServiceURL(opts)
+		if err != nil {
+			return nil, err
+		}
+
+		client, err = azureblob.NewDefaultClient(serviceURL, azureblob.ContainerName(conf.url.Host))
```

Comment:
This internally still looks at the env variables. We need to make sure that if `c.config.AllowHostAccess` we should not access env variables.